proto/pdpb.proto

```diff
@@ -1,303 +1,321 @@
+    rpc GetMembers(GetMembersRequest) returns (GetMembersResponse) {}
+
+    rpc Tso(stream TsoRequest) returns (stream TsoResponse) {}
+
+    rpc Bootstrap(BootstrapRequest) returns (BootstrapResponse) {}
+
+    rpc IsBootstrapped(IsBootstrappedRequest) returns (IsBootstrappedResponse) {}
+
+    rpc AllocID(AllocIDRequest) returns (AllocIDResponse) {}
+
+    rpc GetStore(GetStoreRequest) returns (GetStoreResponse) {}
+
+    rpc PutStore(PutStoreRequest) returns (PutStoreResponse) {}
+
+    rpc StoreHeartbeat(StoreHeartbeatRequest) returns (StoreHeartbeatResponse) {}
+
+    rpc RegionHeartbeat(RegionHeartbeatRequest) returns (RegionHeartbeatResponse) {}
+
+    rpc GetRegion(GetRegionRequest) returns (GetRegionResponse) {}
+
+    rpc GetRegionByID(GetRegionByIDRequest) returns (GetRegionResponse) {}
+
+    rpc AskSplit(AskSplitRequest) returns (AskSplitResponse) {}
+
+    rpc ReportSplit(ReportSplitRequest) returns (ReportSplitResponse) {}
+
+    rpc GetClusterConfig(GetClusterConfigRequest) returns (GetClusterConfigResponse) {}
+
+    rpc PutClusterConfig(PutClusterConfigRequest) returns (PutClusterConfigResponse) {}
+}
+
+message RequestHeader {
+    // cluster_id is the ID of the cluster which be sent to.
+    uint64 cluster_id = 1;
+}
+
+message ResponseHeader {
+    // cluster_id is the ID of the cluster which sent the response.
+    uint64 cluster_id = 1;
+    Error error = 2;
+}
+
+enum ErrorType {
+    OK = 0;
+    UNKNOWN = 1;
+    NOT_BOOTSTRAPPED = 2;
+    STORE_TOMBSTONE = 3;
+}
+
+message Error {
+    ErrorType type = 1;
+    string message = 2;
 }
 
 message TsoRequest {
-    optional uint32 count = 1 [(gogoproto.nullable) = false];
+    RequestHeader header = 1;
+
+    uint32 count = 2;
 }
 
 message Timestamp {
-    optional int64 physical = 1 [(gogoproto.nullable) = false];
-    optional int64 logical  = 2 [(gogoproto.nullable) = false];
+    int64 physical = 1;
+    int64 logical = 2;
 }
 
 message TsoResponse {
-    optional uint32 count        = 1 [(gogoproto.nullable) = false];
-    optional Timestamp timestamp = 2 [(gogoproto.nullable) = false];
+    ResponseHeader header = 1;
+
+    uint32 count = 2;
+    Timestamp timestamp = 3;
 }
 
 message BootstrapRequest {
-    optional metapb.Store store   = 1;
-    optional metapb.Region region = 2;
+    RequestHeader header = 1;
+
+    metapb.Store store = 2;
+    metapb.Region region = 3;
 }
 
 message BootstrapResponse {
-    
+    ResponseHeader header = 1;
 }
 
 message IsBootstrappedRequest {
-    
+    RequestHeader header = 1;
 }
 
 message IsBootstrappedResponse {
-    optional bool bootstrapped = 1;
-}
+    ResponseHeader header = 1;
 
-message AllocIdRequest {
+    bool bootstrapped = 2;
+}
 
+message AllocIDRequest {
+    RequestHeader header = 1;
 }
 
-message AllocIdResponse {
-    optional uint64 id             = 1 [(gogoproto.nullable) = false];
+message AllocIDResponse {
+    ResponseHeader header = 1;
+
+    uint64 id = 2;
```

Comment:
`(gogoproto.nullable) = false` is no longer supported in proto3. 
See more: https://github.com/pingcap/kvproto/pull/135#discussion_r104864493